partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/internal/AccumulatorSum.kt

```diff
@@ -1,18 +1,71 @@
 package org.partiql.spi.function.builtins.internal
 
 import org.partiql.spi.types.PType
+import org.partiql.spi.utils.FunctionUtils.checkIsNumberType
+import org.partiql.spi.utils.FunctionUtils.nullToTargetType
+import org.partiql.spi.utils.NumberUtils.add
+import org.partiql.spi.utils.NumberUtils.toTargetType
 import org.partiql.spi.value.Datum
+import java.math.BigDecimal
 
+// TODO docs + further cleanup
```

Comment:
(self-review) attempted to fix overflow and typing behavior of AVG and SUM in [`c841580` (#1728)](https://github.com/partiql/partiql-lang-kotlin/pull/1728/commits/c84158086c2de1eead1e1d643c35ed48012b585b). Still have to verify and do some further cleanup here and in `NumberUtils`